inspirehep/modules/records/utils.py

```diff
@@ -54,16 +61,22 @@ def get_detailed_template_from_record(record):
 
 
 @contextmanager
-def file_resolving(fpath):
+def open_url_or_path(fpath):
     known_schemes = ['http', 'https']
-    if urlparse(fpath).scheme in known_schemes:
-        with closing(requests.get(url=fpath, stream=True)) as req:
-            if req.status_code == 200:
-                req.raw.decode_content = True
-                yield req.raw
-    else:
-        try:
+    try:
+        if urlparse(fpath).scheme in known_schemes:
+            with closing(requests.get(url=fpath, stream=True)) as resp:
+                if resp.status_code == 200:
```

Comment:
you don't have to implement all of this error handling, you can use `raise_for_status` as explained here: http://docs.python-requests.org/en/master/user/quickstart/#response-status-codes.

Reply:
so I can just first raise and then just do my logic after?